src/main/java/com/odysseusinc/arachne/datanode/service/impl/DataSourceServiceImpl.java

```diff
@@ -354,4 +348,12 @@ public JsonResult unpublishAndDeleteOnCentral(Long dataSourceId) {
 
         return dataSourceRepository.findAllByCentralIdIsNull().collect(Collectors.toList());
     }
+
+    @Override
+    public boolean isDatasourceNameUnique(String name, Long dataSourceId) {
+
+        return dataSourceRepository.findByName(name)
```

Comment:
Probably `countByName` would be better and faster?

Reply:
count will not work here because I need to check update as well - 
so logic here find Datasource by the name and allow to save if it has the same ID:

return dataSourceRepository.findByName(name)
                .map(ds -> ds.getId().equals(dataSourceId))
                .orElse(true);

If to change to count then update will not work

Reply:
But `countByNameAndNotId` works?

Reply:
I do not think that it will work for the new entity case - it has Id Null and it will cause JPA exception ID cannot be null something.... So it would requre 2 method countByName and countByNameAndId
and logic like int count = (entity.id ==null)? repo.countByName(entity.name):repo.countByNameAndId(entity.name, entity.id)

but such a code does not look good for me.

Reply:
In contrast, my concern is the following:
`DataSource` is a heavy object with multiple fields and at least one relation,
why we need to restore completed object graph from database while we require just a single number?
When You don't like ternary operator you always can avoid it with strategy or with implementing a functional way.

Reply:
I have changed mostly because I want to stop this discussion, but I do not believe that couple of extra selects by the primary key can somehow slow down the application.